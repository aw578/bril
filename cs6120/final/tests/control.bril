@main {
  .entry:
    a0: int = const 1;
    b0: int = const 3;
    cond: bool = const true;
    br cond .block1 .block2;
  .block1:
    x1: int = add a0 b0;
    jmp .merge;
  .block2:
    x2: int = add a0 b0;
    jmp .merge;
  .merge:
    y0: int = add a0 b0;
}