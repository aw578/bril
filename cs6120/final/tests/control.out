Congruence classes for function main:
Class undef: ['x1.init', 'x2.init']
Class const 1: ['a0.0']
Class const 3: ['b0.0']
Class const True: ['cond.0']
Class add: ['x1.0', 'y0.0', 'x2.0']
Class get: ['x1.1', 'x2.1']
